src/Symfony/Component/DependencyInjection/Compiler/PassConfig.php

```diff
@@ -88,7 +89,6 @@ public function __construct()
         ]];
 
         $this->afterRemovingPasses = [[
-            new CheckExceptionOnInvalidReferenceBehaviorPass(),
```

Comment:
Are we sure that services removed by previous passes (defied in $this->removingPasses or by user).
Won't trigger such exception?

I don't know if it make send to duplicate the pass, in before-removing + after-removing. But I believe cheking invalid reference should be performed after removal of services.

Reply:
> Are we sure that services removed by previous passes (defied in $this->removingPasses or by user).
> Won't trigger such exception?
sounds like possible situation. but there is no suitable test case to check

can I give a try to add some test cases to check the need to duplicate the `CheckExceptionOnInvalidReferenceBehaviorPass` in before-removing + after-removing sections?

Reply:
I agree with @jderusse, this can break apps with incomplete but unused services.
What about moving this pass just after `RemoveUnusedDefinitionsPass`?

Reply:
looks better, thank you